### DISCUSSION TITLE
Issue Connecting UE to gNB and/or UE does not stay connected

### DISCUSSION THREAD

**Discussion by oliverxsch** [srsran/srsRAN_Project#336] [General Help]
Hello srsRAN community,

I'm encountering difficulties connecting the UE to the gNB in my setup. 
See https://github.com/srsran/srsRAN_Project/issues/315.
I started this disscussion because im not sure if its an issue or something else.
Sometimes im actually able to connect the UE to the gNB (with no change in the configuration). But shortly afterwards the connection is released. The logs and configs in the appendix are for the case when i'm actually able to connect.
If i can't connect my ue console looks simular to this: https://github.com/srsran/srsRAN_Project/issues/199
Setup:

First PC: 
Specs: i5 8500, 16GB Ram
OS: Ubuntu 22.04.03 LTS
UHD 3.15 LTS
Newest srsRAN Project and newest Open5Gs.
Connected via a 10G-Link to a USRP X310
"gNB"

Second PC:
Specs: i5 8500, 8GB Ram
OS: Ubuntu 22.04.03 LTS
UHD 4.1.0.5-3
Newest srsRAN-4G Project.
Connected via a 10G-Link to another USRP X310
"UE"

Both USRPs are connected to an octoclock for their sync and clock.

Both USRPs are connected via cables not antennas with 6db attenuators.

Configuration and Logs (for the case when the UE actually connects, but releases the connection after a few seconds.)
Logs for PHY Level gnb are in debug, everything else is in info.
[logs and conf ping no answer.zip](https://github.com/srsran/srsRAN_Project/files/13459944/logs.and.conf.ping.no.answer.zip)

Console logs:
gNB:
root@Dell-5060-2:/home/inets/oliver-bc-thesis/srsRAN_Project/build# sudo gnb -c /home/inets/oliver-bc-thesis/srsRAN_Project/configs/gnb_srsUE.yml
Lower PHY in dual executor mode.

--== srsRAN gNB (commit 374200dee) ==--

Connecting to AMF on 127.0.0.5:38412
Available radio types: uhd and zmq.
Making USRP object with args 'type=x300,addr=192.168.43.2,num_recv_frames=64,num_send_frames=64,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000'
[INFO] [UHD] linux; GNU C++ version 11.4.0; Boost_107400; UHD_3.15.0.0-74-ge35f66e8
[INFO] [LOGGING] Fastpath logging disabled at runtime.
[INFO] [X300] X300 initialization sequence...
[INFO] [X300] Maximum frame size: 8000 bytes.
[INFO] [X300] Radio 1x clock: 184.32 MHz
[INFO] [GPS] Found an internal GPSDO: LC_XO, Firmware Rev 0.929a
[INFO] [0/DmaFIFO_0] Initializing block control (NOC ID: 0xF1F0D00000000000)
[INFO] [0/DmaFIFO_0] BIST passed (Throughput: 1314 MB/s)
[INFO] [0/DmaFIFO_0] BIST passed (Throughput: 1300 MB/s)
[INFO] [0/Radio_0] Initializing block control (NOC ID: 0x12AD100000000001)
[INFO] [0/Radio_1] Initializing block control (NOC ID: 0x12AD100000000001)
[INFO] [0/DDC_0] Initializing block control (NOC ID: 0xDDC0000000000000)
[INFO] [0/DDC_1] Initializing block control (NOC ID: 0xDDC0000000000000)
[INFO] [0/DUC_0] Initializing block control (NOC ID: 0xD0C0000000000000)
[INFO] [0/DUC_1] Initializing block control (NOC ID: 0xD0C0000000000000)
Cell pci=1, bw=10 MHz, dl_arfcn=368500 (n3), dl_freq=1842.5 MHz, dl_ssb_arfcn=368410, ul_freq=1747.5 MHz

==== gNodeB started ===
Type <t> to view trace
t

-----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4601   15   1   24    15k   22    0   0%      0 |   0.9   14    21k   12   10  45%      0
   1 4601   15   1   27    576    1    0   0%      0 |  18.8   24   5.0k    2    1  33%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0

-----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1    0      0    0    0   0%      0 |   n/a    0      0    0    0   0%      0
   
   
   ---- here i started to ping the ue ---
   
   
   1 4601   15   1   28      0    0    1 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   26      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   24      0    0    7 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   23      0    0    8 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   21      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   12 100%      0 |   n/a    0      0    0    0   0%      0

-----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4601   15   1   20      0    0   12 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   13 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   11 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   12 100%     90 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    8 100%     90 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    7 100%     90 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   14 100%     90 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   12 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%      0 |   n/a    0      0    0    0   0%      0

-----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4601   15   1   20      0    0   11 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    8 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   12 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   12 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0   10 100%      0 |   n/a    0      0    0    0   0%      0

-----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4601   15   1   20      0    0    8 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%      0 |   n/a    0      0    0    0   0%      0
   1 4601   15   1   20      0    0    9 100%     90 |   n/a    0      0    0    0   0%      0

UE:
inets@Dell-3060-5:~/Desktop/srsRAN_4G/build$ sudo srsue ue_web.conf
[sudo] password for inets: 
Active RF plugins: libsrsran_rf_uhd.so libsrsran_rf_zmq.so
Inactive RF plugins: 
Reading configuration file ue_web.conf...

Built in Release mode using commit fa56836b1 on branch master.

Opening 1 channels in RF device=uhd with args=clock=external,sync=external
Supported RF device list: UHD zmq file
[INFO] [UHD] linux; GNU C++ version 11.2.0; Boost_107400; UHD_4.1.0.5-3
[INFO] [LOGGING] Fastpath logging disabled at runtime.
Opening USRP channels=1, args: sync=external,type=x300,master_clock_rate=184.32e6
[INFO] [UHD RF] RF UHD Generic instance constructed
[INFO] [X300] X300 initialization sequence...
[INFO] [X300] Maximum frame size: 8000 bytes.
[INFO] [GPS] No GPSDO found
[INFO] [X300] Radio 1x clock: 184.32 MHz
Waiting PHY to initialize ... done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=1294
Random Access Complete.     c-rnti=0x4601, ta=9
RRC Connected
PDU Session Establishment successful. IP: 10.45.0.4
RRC NR reconfiguration successful.
RF status: O=0, U=1, L=0
Scheduling request failed: releasing RRC connection...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=8814
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=8974
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9134
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9294
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9454
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9614
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=9774
RF status: O=0, U=1, L=0
qReceived RRC Release

Stopping ..
---  exiting  ---

**Comment by pgawlowicz** [comment]
@oliverxsch any update on this issue?

**Reply by azzukky** [reply]
I am also facing this issue, please does anyone have an idea on this ?

**Reply by oliverxsch** [reply]
I believed this was due to non fitting gains, because with SMA cables it worked better.
But currently im having another issue: https://github.com/srsran/srsRAN_Project/issues/480
